src/menus/icon-menu.jsx

```diff
@@ -162,10 +164,12 @@ let IconMenu = React.createClass({
   _handleItemTouchTap(e, child) {
     let isKeyboard = Events.isKeyboard(e);
 
-    this._timeout = setTimeout(() => {
-      this.close(isKeyboard);
-    }, this.props.touchTapCloseDelay);
-
+    if (this.props.closeAfterTap) {
+      this._timeout = setTimeout(() => {
+        this.close(isKeyboard);
+      }, this.props.touchTapCloseDelay);
+    }
+      
     if (isKeyboard) {
       this.refs[this.state.iconButtonRef].setKeyboardFocus();
     }
```

Comment:
Also, I think all of the `isKeyboard` logic should be inside the the new `if` block.